harness/determined/deploy/local/cluster_utils.py

```diff
@@ -123,37 +125,56 @@ def master_up(
         restart_policy = "unless-stopped"
     else:
         restart_policy = "no"
-    if auto_bind_mount:
-        bind_mount = auto_bind_mount
-    else:
-        bind_mount = str(Path.home())
-    if no_auto_bind_mount:
-        bind_mount = ""
 
     env = {
         "INTEGRATIONS_HOST_PORT": str(port),
-        "DET_MASTER_CONFIG": str(master_config_path),
         "DET_DB_PASSWORD": db_password,
         "IMAGE_REPO_PREFIX": image_repo_prefix,
         "DET_VERSION": version,
         "DET_RESTART_POLICY": restart_policy,
-        "DET_AUTO_BIND_MOUNT": bind_mount,
     }
 
-    # When master config yaml is provided, we don't provide our own storage path
-    # as we expect the yaml to specify checkpoint_storage.
+    # Some cli flags for det deploy local will cause us to write a temporary master.yaml.
+    master_conf = {}
+    make_temp_conf = False
+
     if master_config_path is not None:
-        master_config_path = Path(master_config_path).resolve()
-        mount_yaml = Path(__file__).parent.joinpath("mount.yaml").resolve()
-        extra_files.append(str(mount_yaml))
+        with master_config_path.open() as f:
+            master_conf = yaml.safe_load(f)
     else:
-        storage_yaml = Path(__file__).parent.joinpath("storage.yaml").resolve()
-        extra_files.append(str(storage_yaml))
+        master_conf["checkpoint_storage"] = {
+            "type": "shared_fs",
+            "host_path": Path(appdirs.user_data_dir("determined")),
+        }
+        make_temp_conf = True
 
+    if storage_host_path is not None:
         if not storage_host_path.exists():
             storage_host_path.mkdir(parents=True)
+        master_conf["checkpoint_storage"] = {
+            "type": "shared_fs",
+            "host_path": str(storage_host_path.resolve()),
+        }
+        make_temp_conf = True
+
+    if auto_work_dir is not None:
+        master_conf.setdefault("task_container_defaults", {})["interactive_workdir"] = str(
+            auto_work_dir.resolve()
+        )
+        make_temp_conf = True
+
+    if make_temp_conf:
+        fd, temp_path = tempfile.mkstemp(prefix="det-deploy-local-master-config-")
+        with open(fd, "w") as f:
+            yaml.dump(master_conf, f)
+        master_config_path = Path(temp_path)
+
+    # This is always true by now, but mypy needs help.
+    assert master_config_path is not None
 
-        env["DET_CHECKPOINT_STORAGE_HOST_PATH"] = str(storage_host_path)
+    mount_yaml = Path(__file__).parent.joinpath("mount.yaml").resolve()
```

Comment:
Isn't this file already deleted in this PR?

Reply:
ah, good catch.